### PR TITLE
1774: Planning view use service

### PR DESCRIPTION
**Pull request by jeppekroghitk**
#### Link to ticket

https://leantime.itkdev.dk/#/tickets/showTicket/1774

#### Description

* Added planningservice file
* Modified planningController to get data from planningService
* Added editorId as a parameter to Issue

#### Checklist

- [ ] My code is covered by test cases.
- [ ] My code passes our test (all our tests).
- [ ] My code passes our static analysis suite.
- [ ] My code passes our continuous integration process.

### CODE REVIEW

**Review by tuj — CHANGES_REQUESTED** [review]

File: assets/controllers/loader_controller.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { Controller } from "@hotwired/stimulus";
+
+/** Toggle value for key. */
+export default class extends Controller {
+
+    connect() {
+        this.element.classList.toggle('loading');
```
Comment (tuj) [inline assets/controllers/loader_controller.js:7]: remove("loading")

File: assets/styles/app.css [hunk 2]
```diff
@@ -405,5 +405,25 @@
         content: '▼';
         @apply pr-4 text-xs;
     }
-
+    div.loading {
```
Comment (tuj) [inline assets/styles/app.css:408]: remove div prefix

File: migrations/Version20240625130220.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<?php
```
Comment (tuj) [inline migrations/Version20240625130220.php:1]: Merge migrations into one

File: src/Controller/PlanningController.php [hunk 4]
```diff
@@ -84,16 +86,8 @@ private function createResponse(Request $request, string $mode): Response
 
         $form->handleRequest($request);
 
-        $service = null;
-
-        if ($form->isSubmitted() && $form->isValid()) {
```
Comment (tuj) [inline src/Controller/PlanningController.php:89]: I think the $planningData = $this->planningService->getPlanningData(); should only be shown after form is submitted and valid

File: src/Service/PlanningService.php [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+            }
+        }
+
+        return $weekIssues;
+    }
+
+    private function processIssuesForWeek(PlanningData $planning, int $week, array $issues): void
+    {
+        foreach ($issues as $issueData) {
+            if ('0' !== $issueData->getStatus()) {
```
Comment (tuj) [inline src/Service/PlanningService.php:147]: What statuses are possible here? '0' is a weird value. The best would be that getStatus() returns an enum

**Review by tuj — CHANGES_REQUESTED** [review]

File: src/Service/PlanningService.php (hunk 5 above)
Comment (jeppekroghitk) [reply src/Service/PlanningService.php:147]: I will implement this on a seperate branch to break it up a bit.

File: src/Service/PlanningService.php [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+        $weeks = $planning->weeks;
+
+        $now = new \DateTime();
+        $currentYear = (int) $now->format('Y');
+        $currentWeek = (int) $now->format('W');
+
+        $weeksOfYear = $this->dateTimeHelper->getWeeksOfYear($currentYear);
+        foreach ($weeksOfYear as $week) {
+            $firstAndLastDateOfWeek = $this->dateTimeHelper->getFirstAndLastDateOfWeek($week);
+            $weekIsSupport = 1 === $week % 4;
```
Comment (tuj) [inline src/Service/PlanningService.php:67]: Move constants to a const variable or create environment variables for them

**Review by tuj — CHANGES_REQUESTED** [review]

File: src/Service/PlanningService.php [hunk 7]
```diff
@@ -0,0 +1,330 @@
...
+    /**
+     * Retrieves the planning data containing the weeks, issues, assignees, and projects.
+     *
+     * @return PlanningData The planning data object
+     */
+    public function getPlanningData(): PlanningData
+    {
+        $planning = new PlanningData();
+        $planning->weeks = $this->buildPlanningWeeks($planning);
+        $allIssues = $this->issueRepository->findAll();
```
Comment (tuj) [inline src/Service/PlanningService.php:38]: This should be changed to a filtered result where only issues with dueDate in the current year are included.

**Review by tuj — CHANGES_REQUESTED** [review]

**Pull request merged**